### PR TITLE
Fix: $this->pass is empty if DATABASE_PWD_ENCRYPTED is used

### PR DESCRIPTION
**Pull request by hregis**
@eldy  important !

### CODE REVIEW

File: htdocs/adherents/class/adherent.class.php [hunk 1]
```diff
@@ -1948,49 +1948,62 @@ function _load_ldap_dn($info,$mode=0)
...
+		if ($this->birth && ! empty($conf->global->LDAP_MEMBER_FIELD_BIRTHDATE))				$info[$conf->global->LDAP_MEMBER_FIELD_BIRTHDATE] = dol_print_date($this->birth,'dayhourldap');
+		if (isset($this->statut) && ! empty($conf->global->LDAP_FIELD_MEMBER_STATUS))			$info[$conf->global->LDAP_FIELD_MEMBER_STATUS] = $this->statut;
+		if ($this->datefin && ! empty($conf->global->LDAP_FIELD_MEMBER_END_LASTSUBSCRIPTION))	$info[$conf->global->LDAP_FIELD_MEMBER_END_LASTSUBSCRIPTION] = dol_print_date($this->datefin,'dayhourldap');
+
+		// Password
+		if ($this->pass && ! empty($conf->global->LDAP_MEMBER_FIELD_PASSWORD))				$info[$conf->global->LDAP_MEMBER_FIELD_PASSWORD] = $this->pass;	// this->pass = mot de passe non crypte
+		if (! empty($conf->global->DATABASE_PWD_ENCRYPTED))
+		{
+			if ($this->pass_indatabase_crypted && ! empty($conf->global->LDAP_MEMBER_FIELD_PASSWORD_CRYPTED))	{
+				$info[$conf->global->LDAP_MEMBER_FIELD_PASSWORD_CRYPTED] = dol_hash($this->pass_indatabase_crypted, 5); // md5 for OpenLdap TODO add type of encryption
```
Comment (eldy) [inline htdocs/adherents/class/adherent.class.php:1989]: pass_indatabase_crypted is the value in database of password "crypted".
If you make dol_hash on it, you crypt the crypted value; Pass should not be crypted twice

**Comment by hregis** [comment]
@eldy 

I have added a type of hash in dol_hash : 

`else if ($type == 5) return '{md5}'.base64_encode(hex2bin($chain)); // For OpenLdap with md5 (based on a md5 encrypted password in base)`

**dolibarr encrypt password:**
`md5("password") = 5f4dcc3b5aa765d61d8327deb882cf99`

**dol_hash('password', 4)**
`"{md5}".base64_encode(mhash(MHASH_MD5,"password")) = {md5}X03MO1qnZdYdgyfeuILPmQ==`

**dol_hash('5f4dcc3b5aa765d61d8327deb882cf99', 5)**
`"{md5}".base64_encode(hex2bin(5f4dcc3b5aa765d61d8327deb882cf99)) = {md5}X03MO1qnZdYdgyfeuILPmQ==`

**Comment by hregis** [comment]
@eldy 
it's ok with md5 and with md5 + MAIN_SECURITY_SALT
it's ok with MAIN_SECURITY_HASH_ALGO = sha1 ( + MAIN_SECURITY_SALT)
but it's not ok with MAIN_SECURITY_HASH_ALGO = sha1md5 !!!

it will be necessary to add a field to store the password ldap !! it's OK for you ?

**Comment by hregis** [comment]
@eldy 
Due to tests I have to create errors, there is no need for modification! meaculpa! :-)

**Pull request closed without merging**